clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -53,6 +53,31 @@ const char *SYCL::Linker::constructLLVMSpirvCommand(Compilation &C,
   return OutputFileName;
 }
 
+void SYCL::constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
```

Comment:
My (non-insisting) suggestion would be to change this function into:
1) accepting the `"--in-file-list"` argument in the form of `InputInfo &`;
2) accepting all other necessary args in `InputInfo`/`StringRef` form;
3) not accepting any `ForeachArgs` container, not requiring `--in-replace` etc.
4) setting `"--in-replace"`, `"--out-replace"` internally to any comfortable value (e.g. `{}`). Or taking the desired `"--*-replace"` values as `StringRef`s.
4) maybe even returning something `unique_ptr<Command>` and letting the caller `addCommand` it instead.

Reply:
1. Seems ok for me.
2. You mean, for example, get command arguments and executable in `StringRef` form too, right?
3. Seems ok for me.
4. Seems ok for me.
5. Not sure about it. For example previous function `constructLLVMSpirvCommand` contains adding command to the `Compilation`. It's ok if I leave it as it?

Reply:
2. `InputCommand` should probably remain as is, sorry for the confusion.
5. You're right, consistency should be retained.

Reply:
It looks like I did all except 4). It's because I don't have an idea which values set to `in-replace` in case when there are multiple inputs for `llvm-foreach`, so it would be nice If you have a suggestion.

Reply:
Ah, I didn't see this coming. Your approach is the best then - maybe wrapping `Filename` into some braces would make `--in-replace` tokens a tad clearer, but please don't bother yourself if you don't think this necessary.